### ISSUE TITLE
BUG: DOF reported in model.fit() seems wrong

### ISSUE THREAD

**Issue by ialxn** [hyperspy/hyperspy#1610]
Dear all

I am fitting an EDS spectra with 2048 data point with an ``order=2`` polynomial background and 9 x-ray lines.
``model.dof.data[0] = 9`` after ``model.fit()`` and after an additional ``model.calibrate_energy_axis()`` it is 18. These numbers do not make sense.

1. case: I have 2 plus 9 (one intensity per line) = 11 free parameters and DOF should be 2048-11=2037
2. case: I have 2 plus 9 (intensities) plus 9 (energies) = 20 free parameters and DOF should be 2028

in both cases (almost) the same ``model.chisq.data[0]`` and  ``model.red_chisq.data[0]`` are reported and they differ by a factor of 2031 which would correspond to 17 fitted parameters.

**Comment by to266** [comment]
Hi there,

First, just to make it clear: 
 - the `model.dof` directly corresponds to the number of parameters that were available for the optimiser to use (we literally just count the number of values that the optimiser returned as parameters, no fancy calculations). This does __not__ include the number of energy channels. I suppose given the wikipedia definition, this should be named `nofp` for "number of fitted parameters", but I will continue using `dof`.
 - Due to the previous definition, `red_chisq` is calculated as `chisq / (number_of_channels - dof - 1)`. I admit I currently do not remember exactly why the "minus 1" is there - I believe it has something to do with "noise term"? Anyway, if you feel strongly about it, feel free to research and propose your correction. Given the usual `number_of_channels` magnitude it should not matter much.

__Why you suddenly get more DOF after calibrating the energy axis:__ the function (with default arguments in particular) does a couple of things:
 1. Fixes the distances between alpha lines (`model.xray_lines`) w.r.t the first one.
 2. Relaxes all alpha lines widths.
 3. Fits 
 4. Undoes (1) and (2).

Importantly, the (2) step adds one more free parameter per line. As the (3) step actually performs the fitting, `model.dof.data[ind]` gets overwritten with that particular fit DOF.

Another thing: `order=2` polynomial has 3 free parameters:
`a_0 + x a_1 + x^2 a_2`

**Comment by ialxn** [comment]
Ok, thanks. Then it was basically just a question of semantics.
